vault/resource_database_secret_backend_connection.go

```diff
@@ -609,6 +622,8 @@ func getDatabaseAPIData(d *schema.ResourceData) (map[string]interface{}, error)
 		setElasticsearchDatabaseConnectionData(d, "elasticsearch.0.", data)
 	case "snowflake-database-plugin":
 		setSnowflakeDatabaseConnectionData(d, "snowflake.0.", data)
+	case "redshift-database-plugin":
+		setDatabaseConnectionData(d, "redshift.0.", data)
```

Comment:
It looks like `setDatabaseConnectionData()` does not include support for [username](https://www.vaultproject.io/api-docs/secret/databases/redshift#username) or [password](https://www.vaultproject.io/api-docs/secret/databases/redshift#password)

Seems to be the same issue for [postgresql](https://www.vaultproject.io/api-docs/secret/databases/postgresql) as well, so the omissions might be intentional.

Reply:
Good catch! That definitely looked like a bug. I confirmed that all DBs using `setDatabaseConnectionData` have username and password fields, and I added support for username and password to that function.